components/scream/src/share/io/scorpio_output.cpp

```diff
@@ -304,12 +303,11 @@ void AtmosphereOutput::register_views()
   // Cycle through all fields and register.
   for (auto const& name : m_fields)
   {
-    auto fmap = m_field_repo->get_field(name, m_grid_name);
+    auto field = m_field_repo->get_field(name, m_grid_name);
     // If the "averaging type" is instant then just need a ptr to the view.
-    auto view_d = fmap.get_view();
+    auto view_d = field.get_view();
```

Comment:
Using `get_view` is dangerous if we're dealing with a subfield. We should put a check on `field`, to make sure it is not a subfield:

```
EKAT_REQUIRE_MSG (field.get_header().get_parent()==nullptr, "Error! Cannot  deal with subfield, for now.");
```

Reply:
yes, good catch.